runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -56,6 +57,27 @@
 
   private Environments() {}
 
+  public static Optional<Environment> getEnvironment(
+      String ptransformId, Components components) {
+    try {
+      PTransform ptransform = components.getTransformsOrThrow(ptransformId);
+      String envId =
+          KNOWN_URN_SPEC_EXTRACTORS
+              .getOrDefault(ptransform.getSpec().getUrn(), DEFAULT_SPEC_EXTRACTOR)
+              .getEnvironmentId(ptransform);
+      if (!Strings.isNullOrEmpty(envId)) {
+        // Some PTransform payloads may have an empty (default) Environment ID, for example a
```

Comment:
The default value in a proto string field is the empty field.

So its better to word this as an "unspecified" environment id.

Reply:
Done.